pyanaconda/installation_tasks.py

```diff
@@ -435,7 +442,15 @@ def run_task(self):
         related machinery.
         """
         if self._task:
-            self._task(*self._task_args, **self._task_kwargs)
+            try:
+                self._task(*self._task_args, **self._task_kwargs)
+            except Exception as e:
+                # If an exception is raised when processing the task, run the exception through
+                # the error handler and raise it back if it does not match any of the exceptions
+                # we can handle.
+                if errors.errorHandler.cb(e) == errors.ERROR_RAISE:
```

Comment:
I'm sure we can find a better way & implement that in a followup PR. But at least it seems to work fine so far in all the cases I have tried & it has been like this since the initial implementation.